valhalla/thor/worker.h

```diff
@@ -113,13 +113,13 @@ class thor_worker_t : public service_worker_t {
   sif::mode_costing_t mode_costing;
 
   // Path algorithms (TODO - perhaps use a map?))
-  BidirectionalAStar bidir_astar;
-  AStarBSSAlgorithm bss_astar;
-  MultiModalPathAlgorithm multi_modal_astar;
-  TimeDepForward timedep_forward;
-  TimeDepReverse timedep_reverse;
+  std::unique_ptr<BidirectionalAStar> bidir_astar;
+  std::unique_ptr<AStarBSSAlgorithm> bss_astar;
+  std::unique_ptr<MultiModalPathAlgorithm> multi_modal_astar;
+  std::unique_ptr<TimeDepForward> timedep_forward;
+  std::unique_ptr<TimeDepReverse> timedep_reverse;
 
-  Isochrone isochrone_gen;
+  std::unique_ptr<Isochrone> isochrone_gen;
```

Comment:
its not obvious to me the benefit of this change, what am i missing?

Reply:
since we have to recreate all of these classes in `thor_worker` constructor (because of new parameter `max_reserved_labels_count`), it's not allowed to write `bidir_astar = BidirectionalAstar(...)` because it will cause copying.

idea: We can leave nonpointers here and explicitly forbid copy-constructor for these classes

Reply:
there are 2 options I see:
1) add move constructor/operator= for each class derived from `PathAlgorithm` and `Dijkstras` and leave all destructors empty (because in case of `move` all states inside classes will be invalid).
2) explicitly forbid copy/move constructors for `PathAlgorithm` and `Dijkstras` (in this case we will be able to access the internal state in destructors) and leave `unique_ptr`s

Reply:
you can just do it in the initializer list. first in the initializer list you set `reserved_label_count`, the you use it to initialize all the other algorithme members